example/src/main/java/com/stripe/example/activity/AmazonPayActivity.kt

```diff
@@ -0,0 +1,138 @@
+package com.stripe.example.activity
+
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Button
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material.Divider
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.themeadapter.material.MdcTheme
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.model.PaymentMethodCreateParams
+
+class AmazonPayActivity : StripeIntentActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            var flow by rememberSaveable { mutableStateOf(AmazonPayFlow.Payment) }
+
+            val isProcessing by viewModel.inProgress.observeAsState(initial = false)
+            val status by viewModel.status.observeAsState(initial = "")
+
+            val canSubmit = remember(flow) {
+                flow == AmazonPayFlow.Payment
+            }
+
+            AmazonPayScreen(
+                flow = flow,
+                isProcessing = isProcessing,
+                canSubmit = canSubmit,
+                status = status,
+                onFlowTypeChanged = {
+                    viewModel.status.postValue("")
+                    flow = it
+                },
+                onButtonPressed = { payWithAmazonPay(flow) },
+            )
+        }
+    }
+
+    private fun payWithAmazonPay(
+        flow: AmazonPayFlow,
+    ) {
+        when (flow) {
+            AmazonPayFlow.Payment -> {
+                val params = PaymentMethodCreateParams.createAmazonPay()
+
+                createAndConfirmPaymentIntent(
+                    country = "US",
+                    paymentMethodCreateParams = params,
+                    supportedPaymentMethods = PaymentMethod.Type.AmazonPay.code
+                )
+            }
+        }
+    }
+}
+
+private enum class AmazonPayFlow {
+    Payment,
+    // TODO-someday: Add PaymentWithFutureUse and Setup
```

Comment:
We don't need a TODO here, we can use jira to track the future changes.

Reply:
gotcha, removed it